app/Http/Controllers/InternetController.php

```diff
@@ -149,16 +153,14 @@ public static function extendUsersInternetAccess(User $user)
 
     public function addMacAddress(Request $request)
     {
+        $this->authorize('create', MacAddress::class);
+
         $validator = Validator::make($request->all(), [
             'comment' => 'required|max:1000',
             'mac_address' => ['required', 'regex:/((([a-fA-F0-9]{2}[-:]){5}([a-fA-F0-9]{2}))|(([a-fA-F0-9]{2}:){5}([a-fA-F0-9]{2})))/i'],
         ]);
         $validator->validate();
 
-        if ($validator->fails()) {
-            return back()->withErros($validator)->withInput();
-        }
-
```

Comment:
Why did you delete this?

Reply:
And the others as well

Reply:
It seems to me that `$validator->validate();` does the same, so if the validator would fails, we return there and would never go into this function. You can check by adding a `die();` call after the `$validator->validate();` line.

Reply:
Yes, but do validate returns the error messages and the input? I think it only returns a template error message. Can you check?

Reply:
Yes, both were returned here.